lib/msf/core/db_manager/vuln.rb

```diff
@@ -262,8 +262,9 @@ def update_vuln(opts)
   ::ActiveRecord::Base.connection_pool.with_connection {
     wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework, false)
     opts[:workspace] = wspace if wspace
-    id = opts.delete(:id)
-    Mdm::Vuln.update(id, opts)
+    v = Mdm::Vuln.find(opts.delete(:id))
+    v.update!(opts)
```

Comment:
We spoke in chat regarding my concern, but I wanted to document it here in the PR. The update now spans two operations, `find` and `update!`. If another update or delete request is processed in between these two the result may not be correct.

Reply:
Noted. I think that the likelihood of this causing a collision is low, but do acknowledge it is possible. This is really a Rails constraint, and since we're looking at moving away from the framework in the future, it will probably get resolved then.